lib/matplotlib/backends/backend_ps.py

```diff
@@ -616,65 +611,43 @@ def draw_text(self, gc, x, y, s, prop, angle, ismath=False, mtext=None):
         if ismath == 'TeX':
             return self.draw_tex(gc, x, y, s, prop, angle)
 
-        elif ismath:
+        if ismath:
             return self.draw_mathtext(gc, x, y, s, prop, angle)
 
-        elif mpl.rcParams['ps.useafm']:
-            self.set_color(*gc.get_rgb())
-
+        if mpl.rcParams['ps.useafm']:
             font = self._get_font_afm(prop)
-            fontname = font.get_fontname()
-            fontsize = prop.get_size_in_points()
-            scale = 0.001 * fontsize
+            scale = 0.001 * prop.get_size_in_points()
 
             thisx = 0
-            last_name = None
-            lines = []
+            last_name = None  # kerns returns 0 for None.
+            xs_names = []
             for c in s:
                 name = uni2type1.get(ord(c), f"uni{ord(c):04X}")
                 try:
                     width = font.get_width_from_char_name(name)
                 except KeyError:
                     name = 'question'
                     width = font.get_width_char('?')
-                if last_name is not None:
-                    kern = font.get_kern_dist_from_name(last_name, name)
-                else:
-                    kern = 0
+                kern = font.get_kern_dist_from_name(last_name, name)
                 last_name = name
                 thisx += kern * scale
-
-                lines.append('%f 0 m /%s glyphshow' % (thisx, name))
-
+                xs_names.append((thisx, name))
                 thisx += width * scale
 
-            thetext = "\n".join(lines)
-            self._pswriter.write(f"""\
-gsave
-/{fontname} findfont
-{fontsize} scalefont
-setfont
-{x:f} {y:f} translate
-{angle:f} rotate
-{thetext}
-grestore
-""")
-
         else:
             font = self._get_font_ttf(prop)
             font.set_text(s, 0, flags=LOAD_NO_HINTING)
             self._character_tracker.track(font, s)
+            xs_names = [(item.x, font.get_glyph_name(item.glyph_idx))
+                        for item in _text_layout.layout(s, font)]
 
-            self.set_color(*gc.get_rgb())
-            ps_name = (font.postscript_name
-                       .encode('ascii', 'replace').decode('ascii'))
-            self.set_font(ps_name, prop.get_size_in_points())
-
-            thetext = '\n'.join(
-                '{:f} 0 m /{:s} glyphshow'
-                .format(item.x, font.get_glyph_name(item.glyph_idx))
-                for item in _text_layout.layout(s, font))
-            self._pswriter.write(f"""\
+        self.set_color(*gc.get_rgb())
+        ps_name = (font.postscript_name
+                   .encode("ascii", "replace").decode("ascii"))
+        self.set_font(ps_name, prop.get_size_in_points())
```

Comment:
I believe that's fine, and in fact that's how `set_font` is supposed to work (its `save` parameter basically assumes that it is setting the font globally (outside of any `gsave` block) so that it know when it needs to reset the font or not.